agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -1607,3 +1607,58 @@ func TestAppMeshCNIPlugin(t *testing.T) {
 
 	assert.Equal(t, 42, exitCode, fmt.Sprintf("Expected exit code of 42; got %d", exitCode))
 }
+
+// TestTrunkENIAttachDetachWorkflow tests that when ENI trunking is enabled, the Trunk ENI is attached
+// when container instance reaches ACTIVE status, and it's detached when the container instance is deregistered
+func TestTrunkENIAttachDetachWorkflow(t *testing.T) {
+	asyncWaitDuration := 30 * time.Second
+
+	// TODO: add more when backend supports more
+	RequireInstanceTypes(t, []string{"c5", "m5"})
+
+	// Enable ENI Trunking account setting
+	putAccountSettingInput := ecsapi.PutAccountSettingInput{
+		Name:  aws.String("awsvpcTrunking"),
+		Value: aws.String("enabled"),
+	}
+	_, err := ECS.PutAccountSetting(&putAccountSettingInput)
+	require.NoError(t, err)
+
+	existingInterfaceCount, err := GetNetworkInterfaceCount()
+	require.NoError(t, err)
+
+	agentOptions := &AgentOptions{
+		EnableTaskENI: true,
+	}
+	os.Setenv("ECS_FTEST_FORCE_NET_HOST", "true")
+
+	agent := RunAgent(t, agentOptions)
+	defer func() {
+		err := agent.StopAgent()
+		assert.NoError(t, err)
+
+		_, err = ECS.DeregisterContainerInstance(&ecsapi.DeregisterContainerInstanceInput{
+			Cluster:           &agent.Cluster,
+			ContainerInstance: &agent.ContainerInstanceArn,
+			Force:             aws.Bool(true),
+		})
+		assert.NoError(t, err)
+
+		// Wait and verify that the Trunk ENI is detached
+		err = WaitNetworkInterfaceCount(existingInterfaceCount, asyncWaitDuration)
+		assert.NoError(t, err)
+
+		agent.TestCleanup()
+	}()
+	// TODO: when merging the changes, change to 1.27.1
+	agent.RequireVersion(">=1.26.0")
+
+	// Wait for container instance to become active
+	err = agent.WaitContainerInstanceActive(asyncWaitDuration)
+	require.NoError(t, err)
+
+	// Expect one more interface to be attached (i.e. the Trunk)
+	interfaceCount, err := GetNetworkInterfaceCount()
+	require.NoError(t, err)
+	assert.Equal(t, existingInterfaceCount + 1, interfaceCount)
```

Comment:
is there any other way to verify this rather than checking count +1 ? like expected trunk interface name, querying etc.? 
(+1 doesn't always guarantee trunk attached right?)

Reply:
i guess i can check this with DescribeContainerInstances response.. will make the change

Reply:
fixed